### PR TITLE
Sample

### PR DESCRIPTION
**Pull request by jeff-hernandez**
This PR adds a sample transform.

#### Examples
```
label_times.sample(n=10) # returns 10 labels
label_times.sample(frac=.5) # returns 50 percent of labes
label_times.sample(n={True: 10, False: 5}) # returns 10 true and 5 false
label_times.sample(frac={True: .1, False: .5}) # return .1 of all true and .5 of all false
```

### CODE REVIEW

File: composeml/label_times.py [hunk 1]
```diff
@@ -226,3 +226,38 @@ def bin(self, bins, quantiles=False, labels=None, right=True):
 
         label_times.transforms.append(transform)
         return label_times
+
+    def sample(self, n=None, frac=None, random_state=None):
```
Comment (kmax12) [inline composeml/label_times.py]: add the doc string

File: composeml/label_times.py [hunk 2]
```diff
@@ -226,3 +226,38 @@ def bin(self, bins, quantiles=False, labels=None, right=True):
...
+        if isinstance(n, int):
+            sample = super().sample(n=n, random_state=random_state)
+            return sample
+
+        if isinstance(n, dict):
+            sample_per_label = []
+            for label, n, in n.items():
+                is_label = self[self.name].eq(label)
```
Comment (kmax12) [inline composeml/label_times.py]: can these two lines can just be 

`label  = self[self[self.name] == label]`

i don't think you need to do `.loc` or `index`

File: composeml/label_times.py [hunk 3]
```diff
@@ -226,3 +226,38 @@ def bin(self, bins, quantiles=False, labels=None, right=True):
...
+                label = self.loc[self.index[is_label]]
+                sample = label.sample(n=n, random_state=random_state)
+                sample_per_label.append(sample)
+
+            labels = pd.concat(sample_per_label, axis=0, sort=False)
+            return labels
+
+        if isinstance(frac, float):
+            n = int(len(self) * frac)
+            sample = super().sample(n=n, random_state=random_state)
```
Comment (kmax12) [inline composeml/label_times.py]: pandas `sample` has a `frac` parameter: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sample.html

can we use that?

File: composeml/label_times.py (hunk 1 above)
Comment (jeff-hernandez) [reply composeml/label_times.py]: added

File: composeml/label_times.py (hunk 2 above)
Comment (jeff-hernandez) [reply composeml/label_times.py]: refactored the lines

File: composeml/label_times.py (hunk 3 above)
Comment (jeff-hernandez) [reply composeml/label_times.py]: yes, refactored to use pandas parameter

File: composeml/label_times.py [hunk 4]
```diff
@@ -226,3 +226,45 @@ def bin(self, bins, quantiles=False, labels=None, right=True):
...
+
+        Args:
+            n (int or dict) : Sample labels by count. A dictionary returns
+                the number of samples per label. Cannot be used with frac.
+            frac (float or dict) : Sample labels by fractions. A dictionary returns
+                the fraction of the count per label. Cannot be used with n.
+            random_state (int) : Seed for the random number generator.
+
+        Returns:
+            LabelTimes : Random sample of labels.
```
Comment (kmax12) [inline composeml/label_times.py]: can we add examples to the docstring like we do the other transforms?

Comment (jeff-hernandez) [reply composeml/label_times.py]: yes, added examples

**Pull request merged**